virttest/utils_net.py

```diff
@@ -2379,6 +2381,58 @@ def cleanup(self):
         self.close_session()
 
 
+def ieee_eui_generator(base, mask, start=0, repeat=False):
+    """
+    IEEE extended unique identifier(EUI) generator.
+
+    :param base: The base identifier number.
+    :param mask: The mask to calculate identifiers.
+    :param start: The ordinal number of the first identifier.
+    :param repeat: Whether use repeated identifiers when exhausted.
+
+    :return generator: The target EUI generator.
+    """
+    def generator():
+        offset = 0
+        while True:
+            out = base + ((start + offset) & mask)
+            yield out
+            offset += 1
+            if offset > mask:
+                if not repeat:
+                    break
+                offset = 0
+
+    return generator()
```

Comment:
It is same to yield out instead of define a `generator` in `ieee_eui_generator`

Reply:
updated, thanks!